packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -2077,23 +2077,58 @@ function warnAboutInvalidUpdatesOnClassComponentsInDEV(fiber) {
   }
 }
 
+export function warnIfNotScopedWithMatchingAct(fiber: Fiber): void {
```

Comment:
This warning should live next to the rest of the `act` warning logic.

Reply:
This needs to be accessible from 2 modules - ReactFiberHooks (while dispatching a state update) and ReactFiberReconciler (when creating a new container).

Reply:
Ah I misunderstood what this warning was. I got confused and thought it was about warning when you interleave multiple acts.